x-pack/plugins/ml/server/models/annotation_service/annotation.ts

```diff
@@ -224,9 +248,15 @@ export function annotationProvider(callAsCurrentUser: LegacyAPICaller) {
       const docs: Annotations = _.get(resp, ['hits', 'hits'], []).map((d: EsResult) => {
         // get the original source document and the document id, we need it
         // to identify the annotation when editing/deleting it.
-        return { ...d._source, _id: d._id } as Annotation;
+        // if original `event` is undefined then substitute with 'user` by default
+        // since annotation was probably generated by user on the UI
+        return { ...d._source, event: d._source?.event ?? 'user', _id: d._id } as Annotation;
```

Comment:
Suggest to make `'user'` like mentioned in other comments above.

Reply:
Updated in https://github.com/elastic/kibana/pull/70198/commits/36a5a445ef9f3c670fc128c9456cd4b53ca30215